### PR TITLE
Clean up v1alpha1 serving for authentication APIs

### PR DESCRIPTION
**Pull request by liggitt**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
kube-apiserver: inactive serving code is removed for authentication.k8s.io/v1alpha1 APIs
```

/sig auth
/kind cleanup

### CODE REVIEW

**Comment by liggitt** [comment]
/assign @enj @deads2k 
follow-up from https://github.com/kubernetes/kubernetes/pull/126977 we forgot to do in 1.32

File: test/integration/auth/selfsubjectreview_test.go [hunk 1]
```diff
@@ -39,7 +38,7 @@ import (
 
 func TestGetsSelfAttributes(t *testing.T) {
 	// KUBE_APISERVER_SERVE_REMOVED_APIS_FOR_ONE_RELEASE allows for APIs pending removal to not block tests
-	// TODO: Remove this line once authentication v1alpha1 types to be removed in 1.32 are fully removed
+	// TODO: Remove this line when oldest emulation version is 1.35, along with removal of v1beta1 SelfSubjectReview (unservable by default in 1.33, still servable via envvar in 1.34)
```
[the remark this answers is not in the document]
Comment (liggitt) [reply test/integration/auth/selfsubjectreview_test.go:41]: does release refer to the emulated release or the binary release? the envvar was invented before that was a thing :)

[the remark this answers is not in the document]
Comment (Jefftree) [reply test/integration/auth/selfsubjectreview_test.go:41]: well before compatibility versions were introduced, binaryRelease == emulatedRelease. Even in that scenario, I don't think SelfSubjectReview is servable in 1.34 via envvar. The envvar should only allow it to be served in 1.33.

[the remark this answers is not in the document]
Comment (liggitt) [reply test/integration/auth/selfsubjectreview_test.go]: oh, right... I was off by one... updated

**Comment by sftim** [comment]
Changelog suggestion
```diff
-kube-apiserver: inactive serving code is removed for authorization.k8s.io/v1alpha1 APIs
+kube-apiserver: removed inactive serving code, associated with authorization.k8s.io/v1alpha1 APIs
```

**Review by Jefftree — COMMENTED** [review]
LGTM

nit: Title & release note should be authentication APIs rather than authorization.

**Pull request merged**